### ISSUE TITLE
DebuggerNonUserCode attribute and XAML (WPF) Intellisense

### ISSUE THREAD

**Issue by r-work** [CommunityToolkit/dotnet#107]
Not sure if this is the right place, but here it goes.

Generated commands do not show up in XAML intellisense.

Here is ViewModel example:
```C#
public partial class MainWindowViewModel
{
     public System.Windows.Input.ICommand TestCommand { get; }

     [ICommand]
     private void ButtonClick(MainWindowViewModel model)
     {
            MessageBox.Show("test");
     }
}
```

And you can set it XAML:
```xaml
<Window.DataContext>
    <local:MainWindowViewModel />
</Window.DataContext>
```
Intellisense shows **TestCommand** but not **ButtonClickCommand**, and if I decorate **TestCommand** with **DebuggerNonUserCode** as well, that will disappear from Intellisense as well.

![image](https://user-images.githubusercontent.com/68378520/151997289-cf577cfa-f165-4a8d-8ae9-627e229bd83c.png)

Is there a workaround?

**Comment by BretJohnson** [comment]
I believe the deal here is that the Community toolkit MVVM tooling generates the attribute, here: https://github.com/CommunityToolkit/dotnet/blob/0936b58854f1d2a1120fff2ac82d21eb430db95e/CommunityToolkit.Mvvm.SourceGenerators/Input/ICommandGenerator.Execute.cs#L199. And thus the question is what it should generate instead, to have events show in Intellisense but have the debugger skip over it.  Maybe just `DebuggerStepThroughAttribute` would achieve the desired result? I'm not sure. Probably worth a try.

**Comment by TheBlubb14** [comment]
> I have a feeling that may be by design by Roslyn to "hide" these because our XAML IntelliSense doesn't do anything with that attribute. Your best bet may be to figure out a way to avoid getting that attribute applied to your generated code.

The source code generator will generate a command property with `System.Diagnostics.DebuggerNonUserCodeAttribute` when using the `CommunityToolkit.Mvvm.Input.ICommandAttribute` on a method.
With this `DebuggerNonUserCodeAttribute` present on all automatic generated commands we do not get any xaml intellisense for them.

So the `DebuggerNonUserCodeAttribute` should not be generated for such `CommunityToolkit.Mvvm.Input.IRelayCommand` properties or atleast an option for it to disable it

**Comment by mgoertz-msft** [comment]
[DebuggerHidden](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.debuggerhiddenattribute?view=net-6.0) may be another one worth trying

**Comment by Sergio0694** [comment]
Hey @r-work, thank you for reporting this!
And thank you @mgoertz-msft and @BretJohnson for chiming in! 🙂

I can repro the same issue with the XAML IntelliSense on UWP as well. I checked the generators for `[RegexGenerator]`, `[DllImportGenerator]` and `[JsonSerializable]` (from .NET 6 and .NET 7) for comparison, and in fact none of them are emitting `[DebuggerNonUserCode]`. I'm fine with removing that for those generated members from the MVVM Toolkit that are intended to be used by developers directly, both for consistency with the other generators and to solve this IntelliSense issue. I've opened #132 with the change and I'll be testing a preview package locally to ensure this does in fact solve the issue for me too 😊
